tests/pulse/test_transmon.py

```diff
@@ -14,16 +14,300 @@
 """
 Unit tests for the HardwareHamiltonian class.
 """
-# pylint: disable=too-few-public-methods,redefined-outer-name
+# pylint: disable=too-few-public-methods,redefined-outer-name,too-many-arguments
 import numpy as np
 import pytest
 
 import pennylane as qml
-from pennylane.pulse import HardwareHamiltonian, transmon_interaction
-from pennylane.pulse.transmon import TransmonSettings, a, ad
+from pennylane.pulse import HardwareHamiltonian, transmon_interaction, transmon_drive
+from pennylane.pulse.transmon import (
+    TransmonSettings,
+    a,
+    ad,
+    AmplitudeAndPhaseAndFreq,
+    _reorder_AmpPhaseFreq,
+)
 
 from pennylane.wires import Wires
 
+
+class TestTransmonDrive:
+    def test_d_neq_2_raises_error(self):
+        """Test that setting d != 2 raises error"""
+        with pytest.raises(NotImplementedError, match="Currently only supporting qubits."):
+            _ = transmon_drive(0.5, 0.5, 0.5, [0], d=3)
+
+    def test_attributes_and_number_of_terms(self):
+        """Test that the attributes and the number of terms of the ``ParametrizedHamiltonian`` returned by
+        ``drive`` are correct."""
+
+        Hd = transmon_drive(amplitude=1, phase=2, freq=3, wires=[1, 2])
+
+        assert isinstance(Hd, HardwareHamiltonian)
+        assert Hd.wires == Wires([1, 2])
+        assert len(Hd.ops) == 2
+
+    @pytest.mark.parametrize("amp", 0.5 * np.arange(2, dtype=float))
+    @pytest.mark.parametrize("phase", 0.5 * np.arange(2, dtype=float))
+    @pytest.mark.parametrize("freq", 0.5 * np.arange(2, dtype=float))
+    @pytest.mark.parametrize("t", 0.5 * np.arange(1, 3, dtype=float))
+    def test_all_constant_parameters(self, amp, phase, freq, t):
+        """Test that transmon drive with all constant parameters yields the expected Hamiltonian"""
+        H = transmon_drive(amp, phase, freq, wires=[0])
+
+        def expected(amp, phase, freq, t, wire=0):
+            return (
+                0.5
+                * amp
+                * (
+                    np.cos(phase + freq * t) * qml.PauliX(wire)
+                    - np.sin(phase + freq * t) * qml.PauliY(wire)
+                )
```

Comment:
[personal preference] One could write this function once and call the separate functions before passing their result to this function. I.e. call the above `all_const_expected` and in the other tests write
```
def expected(amp, phase, freq, t, wire=0):
    amp = amp(p[0], t)
    return all_const_expected(amp, phase, freq, t, wire=wire)
```

Reply:
nice suggestion, implemented :)